.github/workflows/setup.yml

```
@@ -1,41 +1,52 @@
 # This is a basic workflow to help you get started with Actions
 
-name: CI
+name: Setup
 
-# Controls when the workflow will run
 on:
   # Triggers the workflow on push or pull request events but only for the "main" branch
   push:
     branches: [ "main" ]
   pull_request:
     branches: [ "main" ]
 
-  # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
   build:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v4
-
+      - uses: actions/checkout@v4 
+      # Downloads all dependencies if they don't exist
       # Runs commands to install xschem, then the PDK files, then open xschem
-        # First steps ensure all dependencies are downloaded
-          # Automatically says yes to all installs (hopefully)
-      - name: Download Tools
+      
+      - name: Install Xschem and Dependencies 
         run: |
-              echo Hello, world!
-              sudo apt-get install tk8.6-dev libxpm-dev tcl8.6-dev libtk-dev libx11-dev -y
+              sudo rm -rf /usr/share/dotnet
+              sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+              sudo apt update
+              sudo apt install -y tk8.6-dev libxpm-dev tcl8.6-dev libx11-dev flex libtool-bin libxaw7-dev libreadline6-dev libcairo2-dev
+              git submodule add https://github.com/google/skywater-pdk.git
+              cd skywater-pdk
+              SUBMODULE_VERSION=latest make submodules -j3 || make submodules -j1
+              make timing
+              git submodule add https://github.com/RTimothyEdwards/magic
+              sudo apt-get install magic
```

```suggestion
              cd $GITHUB_WORKSPACE
              git clone git@github.com:RTimothyEdwards/magic.git
              cd magic
              ./configure
              make database/database.h
              make -j$(nproc)
```
Build commands grabbed from here: <https://github.com/RTimothyEdwards/magic/blob/b7942cffe6de3bac78bef3ee745097ecd72c689f/.github/workflows/main.yml#L35-L37>